frontend/src/context/form-context.tsx

```diff
@@ -0,0 +1,111 @@
+import { submitRecipe } from '../api/recipe'
+import { createContext, PropsWithChildren, useContext, useReducer } from 'react'
+
+type FormValues = {
+    name: string | undefined
+    ingredients: string | undefined
+    instructions: string | undefined
+    cuisineId: number
+    dietId: number
+    difficultyId: number
+    image: string | undefined
+}
+
+const initialFormValues: FormValues = {
+    name: '',
+    ingredients: '',
+    instructions: '',
+    cuisineId: 0,
+    dietId: 0,
+    difficultyId: 0,
+    image: '',
+}
+
+type FormState = {
+    values: FormValues
+}
+
+type Action =
+    | { type: 'reset' }
+    | { type: 'change'; name: string; value: string | number }
```

Comment:
Since we have the `FormValues` type we can narrow down the type of the property `name` from a string to something more specific